src/CodeRunner.ts

```diff
@@ -114,34 +114,23 @@ export class CodeRunner {
      * @return {Promise<void>} Promise of stopping
      */
     async stop(): Promise<void> {
-        this.runId += 1; // ignore messages coming from last run
         this.setState(RunState.Stopping);
-        this.publishEvent({
+        BackendManager.publish({
             type: BackendEventType.End,
-            runId: this.runId,
             data: "User cancelled run", contentType: "text/plain"
         });
         // Since we use workers, the old one must be entirely replaced to interrupt it
-        BackendManager.stopBackend(this.backend);
+        this.backend.interrupt();
         return this.start();
```

Comment:
Comsync handles restarting the worker. Also note that it's no longer necessarily replacing the worker.

Reply:
Replacing is no longer done. But there's probably more parts of start that should no longer be required, so I'll definitely look further into that.